service/src/test/java/uk/gov/hmcts/reform/fpl/utils/JudgeAndLegalAdvisorHelperTest.java

```diff
@@ -246,5 +262,85 @@ private Judge buildJudge() {
             .judgeEmailAddress("dread@example.com")
             .build();
     }
+
+    @Nested
+    class BuildForHearing {
+
+        private static final String JUDGE_CASE_ASSIGNED_LABEL = "Case assigned to: His Honour Judge JudgeLastName";
+        private static final String ANOTHER_JUDGE_LAST_NAME = "anotherJudgeLastName";
+
+        @Test
+        void noHearingAndNoAllocatedJudge() {
+            Optional<JudgeAndLegalAdvisor> actual = underTest.buildForHearing(
+                CaseData.builder()
+                    .allocatedJudge(null)
+                    .build(),
+                NO_HEARING);
+
+            assertThat(actual).isEmpty();
+        }
+
+        @Test
+        void noHearingAndAllocatedJudge() {
+            Optional<JudgeAndLegalAdvisor> actual = underTest.buildForHearing(
+                CaseData.builder()
+                    .allocatedJudge(ALLOCATED_JUDGE)
+                    .build(),
+                NO_HEARING);
+
+            assertThat(actual).isEqualTo(Optional.of(
+                JudgeAndLegalAdvisor.builder()
+                    .allocatedJudgeLabel(JUDGE_CASE_ASSIGNED_LABEL)
+                    .build())
+            );
+        }
+
+        @Test
+        void hearingAndAllocatedJudge() {
+            Optional<JudgeAndLegalAdvisor> actual = underTest.buildForHearing(
+                CaseData.builder()
+                    .allocatedJudge(ALLOCATED_JUDGE)
+                    .build(),
+                Optional.of(element(HearingBooking.builder()
+                    .judgeAndLegalAdvisor(JudgeAndLegalAdvisor.builder()
+                        .judgeLastName(ANOTHER_JUDGE_LAST_NAME)
+                        .judgeTitle(ANOTHER_JUDGE_TITLE)
+                        .build())
+                    .build())));
+
+            assertThat(actual).isEqualTo(Optional.of(
+                JudgeAndLegalAdvisor.builder()
+                    .allocatedJudgeLabel(JUDGE_CASE_ASSIGNED_LABEL)
+                    .judgeLastName(ANOTHER_JUDGE_LAST_NAME)
+                    .judgeTitle(ANOTHER_JUDGE_TITLE)
+                    .useAllocatedJudge("No")
+                    .build())
+            );
+        }
+
+        @Test
+        void hearingAndNoAllocatedJudge() {
+            Optional<JudgeAndLegalAdvisor> actual = underTest.buildForHearing(
+                CaseData.builder()
+                    .build(),
+                Optional.of(element(HearingBooking.builder()
+                    .judgeAndLegalAdvisor(JudgeAndLegalAdvisor.builder()
+                        .judgeLastName(ANOTHER_JUDGE_LAST_NAME)
+                        .judgeTitle(ANOTHER_JUDGE_TITLE)
+                        .build())
+                    .build())));
+
+            assertThat(actual).isEqualTo(Optional.of(
+                JudgeAndLegalAdvisor.builder()
+                    .judgeLastName(ANOTHER_JUDGE_LAST_NAME)
+                    .judgeTitle(ANOTHER_JUDGE_TITLE)
+                    .build())
+            );
+        }
+
```

Comment:
vv minor but a bit of blank space here that can probs be cleaned up

Reply:
👍